packages/flutter_tools/test/runner/flutter_command_test.dart

```diff
@@ -125,8 +125,12 @@ void main() {
       // Crash if called a third time which is unexpected.
       mockTimes = <int>[1000, 2000];
 
-      final DummyFlutterCommand flutterCommand =
-          new DummyFlutterCommand(commandFunction: () async { throwToolExit('fail'); });
+      final DummyFlutterCommand flutterCommand = new DummyFlutterCommand(
+        commandFunction: () async {
+          throwToolExit('fail');
```

Comment:
We could use an annotation that will tell the missing return error code that a method never returns.

Reply:
Yeah. (Or the analyzer could infer such an annotation when it notices that a method can't terminate cleanly, e.g. because all code paths throw or exit?)